pypowsybl/network.py

```diff
@@ -4087,6 +4087,11 @@ def remove_extensions(self, extension_name: str, ids: _Union[str, _List[str]]) -
             ids = [ids]
         _pp.remove_extensions(self._handle, extension_name, ids)
 
+    def attach_new_line_on_line(self, voltage_level_id: str, bbs_or_bus_id: str, line_id: str, new_line_id: str, percent: float = 50.0) -> None:
```

Comment:
Otherwise it looks ok to me:
the only thing I wonder is, should we use dataframes like everywhere else to be able to apply multiple changes with one call ?

Reply:
I see what you mean. But does it make sense to chain modify the network in the same call ? The result of a modification possibly impacting the "applicability" of the next one (for example splitting a line in two to attach a new line).
Anyway after reflection I can probably make use of dataframe for a single modification and supporting key words arguments to pass all parameters and maybe add the parameters for the new line instead of defaulting to the parameters of the existing line.

Reply:
@sylvlecl I just added the support of a dataframe to set the attribute of new line in attach_new_line_on_line. It seems less relevant to do it for attach_voltage_level_on_line (because no adder is used).